### ISSUE TITLE
Add a distinct/unique option to the histogram calculation

### ISSUE THREAD

**Issue by tschmetzer** [qgis/QGIS#47377]
### Feature description

For areal raster images as in example data [raster_stddev_bug.compressed.tif.zip](https://github.com/qgis/QGIS/files/8032938/raster_stddev_bug.compressed.tif.zip) many pixels have the exact same value. In the following example picture all pixel values belonging to the same country have the same value.
![grafik](https://user-images.githubusercontent.com/67278094/154127701-dddb2bbc-72d9-492a-90ec-919f5d8ccd4b.png)

Using the histogram for display is currently only able to show an overall pixel count of the raster but for an optimal display the user might want only unique pixel values to be displayed. This is why I suggest to add an option that considers only unique/distinct values.

### Additional context

This issue relates to the unique feature request for the quantile interpolation: 
https://github.com/qgis/QGIS/issues/47375

**Comment by roya0045** [comment]
That means the graph would only display up to 1 and would only display the spread of values, not quite a histogram and I'm unsure how that would be useful.

**Comment by tschmetzer** [comment]
You're totally right. My specification was not clear enough. I imagined of a sorted list of the pixel values. In the following case it's a randomized list which is why the sorted values seem to be almost on straight line:
![grafik](https://user-images.githubusercontent.com/67278094/154527878-b0d30508-234b-47f9-b26a-b4cb27e1f60b.png)

**Comment by tschmetzer** [comment]
Also pixel value over sorted pixel frequency would be beneficial
